### PR TITLE
Add support for --pip-requirements

### PR DESCRIPTION
**Pull request by murphm8**
Add ability to install arbitrary python packages into the Python 2.7 of the bundle by using a requirements file (ex. `pip freeze`).

### CODE REVIEW

**Comment by murphm8** [comment]
There is work that needs to be done to make sure we invalidate the cache if the provided requirements file has been changed between runs.

**Comment by piraka9011** [comment]
@murphm8 @dirk-thomas are there any updates or should this be closed?

**Comment by murphm8** [comment]
@piraka9011  This is an actively used branch. I am slowly working to merge it.

File: test/installer/test_pip3_installer.py [hunk 1]
```diff
@@ -171,4 +176,67 @@ def test_install_not_required(check_output, check_call):
...
         shutil.rmtree(cache_dir)
-        shutil.rmtree(prefix)
\ No newline at end of file
+        shutil.rmtree(prefix)
+
+
+@patch('subprocess.check_call')
+@patch('subprocess.check_output')
+@patch('builtins.open', mock_open(read_data='rpkg==1.2.3'))
+def test_install_addtional_requirements(check_output, check_call):
```
Comment (samuelgundry) [inline test/installer/test_pip3_installer.py]: nit; test name typo "addtional"

**Pull request merged**